### ISSUE TITLE
`preventDefault` Behaving differently when using `Suspense

### ISSUE THREAD

**Issue by Supertigerr** [solidjs/solid#1178]
### Describe the bug

I don't know if this issue is related to solid router or solid, but When using `event.preventDefault()` for the child element click event of the `Link` tag, While also using `Suspense` and `Lazy`, When clicking on the child element, The browser will still redirect. This does not happen if I don't use `Suspense` and `Lazy`.

Working fine without `Suspense`: https://stackblitz.com/edit/github-imhxsg-f6uejd?file=src%2FApp.tsx
Not working with `Suspense`: https://stackblitz.com/edit/github-imhxsg-x4nwkq?file=src%2FApp.tsx,src%2FHome.tsx

### Your Example Website or App

Links above.

### Steps to Reproduce the Bug or Issue

1. Open the first link, click on `Do Nothing`, It does not redirect you, which is intended and is working fine.
2. Open the second link, which has `Suspense`, do the same thing as step 1, It WILL redirect even with the `event.preventDefault()`.

### Expected behavior

It should not redirect when using `Suspense` and `Lazy`

### Platform

- OS: Windows
- Browser: Chrome
- Version: 104

**Comment by yw662** [comment]
[stopPropagation works though](https://stackblitz.com/edit/github-imhxsg-ezjndf?file=src%2FApp.tsx,src%2FHome.tsx)

However it redirects even though stopPropagation works.
For me it is highly likely the redirection is caused by sth. else.

**Comment by Supertigerr** [comment]
`stopPropagation` does not seem to work when i try it 🤔 What do you mean?

**Comment by yw662** [comment]
yeah `preventDefault` and `stopPropagation` are not the same thing.
However `stopPropagation` does work. I guess `preventDefault` works as well. It is sth. else triggering the redirection.

**Comment by Supertigerr** [comment]
Mhm, sorry i removed the reply because I was dumb xD
I've also noticed that the same thing also happens when adding `preventDefault` to the `Link` tag still triggers the redirect when using it with `Suspense`. Strange.

**Comment by ryansolid** [comment]
As far as I can tell this isn't Suspense specifically. Just that the router event handler order can change compared to Solid's compiled click. The solution we should update the router to use our event delegation mechanism.
